### ISSUE TITLE
Server SCSocket errors are causing worker to crash/respawn

### ISSUE THREAD

**Issue by jondubois** [SocketCluster/socketcluster#111]
To reproduce, emit an object that has a circular reference to itself from the server-side SCSocket.

**Comment by robborden** [comment]
I had a recursive function in the worker which unintentionally was running around 12,000+ times and it caused a stackoverflow error. That caused a circular reference error in the SC code and it looked like it was where it was trying to put the error into a json object.  Perhaps it was trying to json.stringify the callstack and having so many of the same function over and over in the stack caused it to think there was a circular reference?  I'm not well versed enough in JS to know if that's possible.  It was on line 118 of workercluster.js

```javascript
process.send({
      type: 'error',
      data: {
        error: error,
        workerPid: process.pid
      }
    });
```

```javascript
1441992742863 - Origin: Worker (PID 28429)
    [Error] TypeError: Converting circular structure to JSON
    at Object.stringify (native)
    at process.target.send (child_process.js:451:23)
    at EventEmitter.handleError (/var/socketcluster/pubSub/node_modules/socketcluster/lib/workercluster.js:118:13)
    at EventEmitter.emit (events.js:95:17)
    at EventEmitter.SCWorker.errorHandler (/var/socketcluster/pubSub/node_modules/socketcluster/lib/scworker.js:242:8)
    at Domain.<anonymous> (/var/socketcluster/pubSub/node_modules/socketcluster/lib/scworker.js:22:23)
    at Domain.EventEmitter.emit (events.js:95:17)
    at process._fatalException (node.js:249:27)
1441992742866 - Worker 1 exited - Exit code: 0
   >> Worker PID: 28454
1441992743212 - Worker 1 was respawned
```

**Comment by jondubois** [comment]
@robborden Thanks for reporting this! That led me directly to the problem. SC didn't do a very good job at handling worker errors which didn't have a stack trace (such as call stack overflows). This was fixed in v2.3.18.
